### ISSUE TITLE
Mobile typeahead select control

### ISSUE THREAD

**Issue by TomTirapani** [xh/hoist-react#2274]
The mobile Select, built on top of react-select, disables type-to-filter functionality by default. This is because it can be problematic to bring up the device keyboard when the Select is in the lower portion of the screen.

However, for large datasets, it can be quite painful to scroll through the list to find the entry you're looking for. A fully-supported type-to-filter control is necessary to provide a good user experience in mobile apps.

At this point, would like to brainstorm / research potential solutions.

**Comment by TomTirapani** [comment]
I took a stab at implementing a "full-screen" mode for mobile selects (I'm not keen on the name), with the select at the top of the screen. This ensures there is always space to bring up the device keyboard.

https://github.com/xh/hoist-react/pull/2383
https://github.com/xh/toolbox/pull/473

The one wrinkle with the above approach is that its seemingly impossible to automatically bring up the device keyboard in fullscreen mode. Calling input.focus() *outside of a user action* doesn't bring up the keyboard. (https://stackoverflow.com/questions/6837543/show-virtual-keyboard-on-mobile-phones-in-javascript). As such, we can show the list on focus but can't bring up the keyboard.

We could take a different approach, where we make sure we use the same input and only move its position somehow. I tried this and hit a bit of a wall (we need to use a portal to reliably position at the top of the screen), but can reopen investigations if we think bringing up the keyboard is vital.

**Comment by lbwexler** [comment]
could call it "modal"

**Comment by TomTirapani** [comment]
Both PRs have been updated to latest from develop, and are ready for review.

These PRs build on top of react-select, rendering it in a full-screen portal on focus. IIRC, we had also discussed an alternative approach, wherein we would drop react-select in favour of a custom full-screen dialog (e.g. grid of options with a filter input) that would conform to the HoistInput API. Is this approach worth pursuing?
